phoenix-core-client/src/main/java/org/apache/phoenix/execute/MutationState.java

```diff
@@ -1530,7 +1531,7 @@ public List<Mutation> getMutationList() {
                         // REPLAY_ONLY_INDEX_WRITES for first batch
                         // only in case of 1121 SQLException
                         itrListMutation.remove();
-
+                        mutationBatchSuccessCounter++;
```

Comment:
Can't we just use `mutationBatchList.size()`

Reply:
Then to know the successful mutation batches, I would have to first keep total batches initially and have to compute the diff to get batches successfully committed. This I feel is a bit indirect way to compute same thing. WDYT?

Reply:
I feel most of the value is in knowing how many batches we created per commit. That is where we can tune the various parameters to improve performance. If the commit succeeds the `mutationBatchSuccessCounter` will be the same as size of the list. If the commit fails, they might be different but that doesn't seem very interesting from metrics perspective. There is also a `batchCount` variable which seems to be tracking the same thing but

Reply:
Got it. I agree when commit call has failed then our interest will be in knowing why it failed instead of knowing batch count. 
Regarding existing `batchCount` variable, I didn't use it because its shared at MutationState object level. So, if multiple application threads are using one Phoenix connection then they all will increment it and we will end up double counting the number of batches. Though I don't think anyone should use Phoenix connection across multiple application threads but there is nothing stopping them.

I will change to capturing no. of batches computed per commit call. Thanks